### DISCUSSION TITLE
2004A display troubleshooting, i2s?

### DISCUSSION THREAD

**Discussion by jackson4242** [probonopd/MiniDexed#158] [General]
I am super excited about this project, sadly many elements of it are very new to me. 

As of now the backlight of my display comes on but that is it, no text is displayed.  

I want to determine how much is working but there are so many variables I am not really sure where to start.  

When creating a sd card, is it necessary to have the sysex/voice directories setup with properly named .syx files to create a sound?  If I wanted to just make sure the software is running, which feels like the right first step, could I do that and detach the display/rotary encoder and hook it up to a midi device (presently the only thing I have with midi over usb is a poly d) and at least be able to confirm that I have that much correct?  Is it dependent on having those directories and files to do anything?

I saw a video somewhere that had a monitor attached via hdmi, do I need to modify any of the config files to do this?  I guess I would feel a lot better just knowing the software is running, and that would give me the encouragement I need to dive deeper and figure out what I am doing wrong.  

I love the work everyone is putting in to this and can't wait to have it up and running!!  Thank you!!

**Comment by craigyjp** [comment]
You don't need the sysex/voices folder initially, it will work without it. You just get the standard 32 voices of ROM 1A that the DX7 came with. 

As for your display, do you have a contrast pot connected? Is it breadboarded, are all the connections good?

**Reply by jackson4242** [reply]
I don't have a pot connected but I am pretty sure there is contrast control on the display I found.  I adjusted it to be sure I could see any text that might be displayed.  Maybe the display I found is the problem?  I connected the GND, VCC, SDA and SCL to the appropriate places on the RPI, I only used a breadboard for the power, with what I have on hand that made sense, but I am not against ordering any parts necessary to do a better job. 
![rotated](https://user-images.githubusercontent.com/104287789/164958194-46860e98-ec52-43f2-8f73-4beb79c82846.jpg)

**Reply by craigyjp** [reply]
I didn't think an I2c display was supported, parallel only, you would have to remove the I2c adaptor and connect the data wires plus power required for the display.

**Reply by probonopd** [reply]
Correct, i2c displays are not supported (yet?). But you don't have to remove the black board. just connect to its connectors at the very top.

**Reply by probonopd** [reply]
You notice that your 2004A display is not the type MiniDexed needs? MiniDexed is currently calling for a 16x2 character display. You might still get yours to work if the command set is similar (which wouldn't surprise me) beut it might be necessary to do some code changes.

**Reply by probonopd** [reply]
> I don't have a pot connected b

The blue thing on the black board is your pot. It should adjust the contrast.

**Reply by jackson4242** [reply]
I tried hooking up the display via the pins on the front, they are just a pinch too short for it to feel nice, and after a couple tries I still am not seeing anything, so at my skill level I think probably the simplest solution is to save this display for another project and order one that I know will work.  Especially considering how inexpensive they are.  I really appreciate all the help, and am pretty stoked that I can at least use it now, even without a display!!

**Comment by probonopd** [comment]
> I saw a video somewhere that had a monitor attached via hdmi, do I need to modify any of the config files to do this? I guess I would feel a lot better just knowing the software is running, and that would give me the encouragement I need to dive deeper and figure out what I am doing wrong.

You don't need to configure anything for that. Just attach a display.
But the more important question is: Do you get sound out of it? I'd start with the headphone jack on the Raspberry Pi, even though it has the worst quality of all options at least there is nothing that can be wired up wrong. Once that works, change to HDMI or i2s for better sound.

**Reply by jackson4242** [reply]
Thank you, I did achieve sound!!  I tried using hdmi sound, I didn't use a transistor but had a monitor attached and didn't get sound on a RPI 4 or 400, but I suppose if I can't figure that out that is a different thread.  I did switch to HDMI in the settings.  Headphone jack on the 4 works great though.  I really appreciate all the help!!
